reactor-netty-core/src/main/java/reactor/netty/transport/MicrometerEventLoopMeterRegistrar.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2019-2021 VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.transport;
+
+import io.micrometer.core.instrument.Gauge;
+import io.netty.channel.EventLoop;
+import io.netty.util.concurrent.SingleThreadEventExecutor;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static reactor.netty.Metrics.EVENT_LOOP_PREFIX;
+import static reactor.netty.Metrics.NAME;
+import static reactor.netty.Metrics.PENDING_TASKS;
+import static reactor.netty.Metrics.REGISTRY;
+import static reactor.netty.Metrics.STATUS;
+
+/**
+ * Registers gauges for a given @{link {@link EventLoop}.
+ *
+ * Every gauge uses thread name and state as tags.
+ *
+ * @author Pierre De Rop
+ * @since 1.0.14
+ */
+public final class MicrometerEventLoopMeterRegistrar {
+
+	final static MicrometerEventLoopMeterRegistrar INSTANCE = new MicrometerEventLoopMeterRegistrar();
+
+	private final ConcurrentMap<String, EventLoop> cache = new ConcurrentHashMap<>();
+
+	private MicrometerEventLoopMeterRegistrar() {}
+
+	void registerMetrics(EventLoop eventLoop) {
+			if (eventLoop instanceof SingleThreadEventExecutor) {
+				SingleThreadEventExecutor singleThreadEventExecutor = (SingleThreadEventExecutor) eventLoop;
+
+				cache.computeIfAbsent(singleThreadEventExecutor.threadProperties().name(), key -> {
+					Gauge.builder(EVENT_LOOP_PREFIX + PENDING_TASKS, singleThreadEventExecutor::pendingTasks)
+							.description("Event loop pending scheduled tasks.")
+							.tag(NAME, singleThreadEventExecutor.threadProperties().name())
+							.tag(STATUS, singleThreadEventExecutor.threadProperties().state().name())
```

Comment:
I wonder about adding the status to the tags, since that's something that would change overtime... to me it sounds more like a separate Gauge ?

Reply:
Indeed, sounds like adding the event executor state in the STATUS tag is not necessary. I kept it from what was suggested in initial [1398](https://github.com/reactor/reactor-netty/issues/1398) issue, but I do agree that the status is probably unrelated here. I'll update the PR and will remove the STATUS tag.
Thanks for the review !